sending-a-simple-message-to-a-service/README.md

```diff
@@ -0,0 +1,325 @@
+
+6. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and responds to the request. Use `/healthcare` as the service path (or the context) of the service which is attached to the listener listening on port `port`. Define an HTTP resource that allows the `GET` operation on resource path `/querydoctor` and accepts the `category` (corresponding to the specialization) as a path parameter.
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function get querydoctor/[string category]() {
+        
+    }
+}
+```
+
+7. Implement the logic to retrieve and respond with relevant details.
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function get querydoctor/[string category]() 
+            returns Doctor[]|http:NotFound|http:InternalServerError {
+        log:printInfo("Retrieving information", specialization = category);
+        
+        Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+        if resp is Doctor[] {
+            return resp;
+        }
+
+        if resp is http:ClientRequestError {
+            return <http:NotFound> {body: string `category not found: ${category}`};
+        }
+
+        return <http:InternalServerError> {body: resp.message()};
+    }
+}
+```
+
+- The `log:printInfo` statement [logs](https://ballerina.io/learn/by-example/#log) information about the request.
+
+```ballerina
+log:printInfo("Retrieving information", specialization = category);
+```
+
+- The call to the backend is done using a remote method call expression (using `->`) which distinguishes network calls from normal method calls. [Client data binding](https://ballerina.io/learn/by-example/http-client-data-binding/) is used to directly try and bind the JSON response on success to the expected array of records.
+
+```ballerina
+Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+```
+
+- Use the `is` check to decide the response based on the response to the client call. If the client call was successful and the respond payload was an array of `Doctor`s (as expected) directly return the array from the resource. If the request failed, send a "NotFound" response if the client call failed with a 4xx status code or return an "InternalServerError" response for other failures.
+
+```ballerina
+log:printInfo("Retrieving information", specialization = category);
+
+Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+if resp is Doctor[] {
+    return resp;
+}
+
+log:printError("Retrieving doctor information failed", resp);
+if resp is http:ClientRequestError {
+    return <http:NotFound> {body: string `category not found: ${category}`};
+}
+
+return <http:InternalServerError> {body: resp.message()};
+```
+
+#### Complete source
+
+You have successfully developed the required service.
```

Comment:
```suggestion
You have successfully developed the required service below.
```

Reply:
What does this mean? We have successfully developed the service by following the steps mentioned above. The complete source code (which was developed already) is what is available below.

Reply:
Then, we need to change the sentence to that: 
`Below is the complete source code of the service, which you developed above.`

Reply:
Isn't it just a lot more words to say the same thing? The title + the line we have at the moment convey the same idea?

Reply:
issue here is we are saying you developed a service and showing some code snippet. We need to connect the 2 properly.

Reply:
We can get rid of this sentence entirely. Since the title indicates what it is.

Reply:
Will move this sentence to the previous section.

Reply:
Updated.